.github/workflows/eslint.yml

```diff
@@ -1,32 +1,55 @@
-name: ESLint
+name: FE check
 
 on:
   pull_request:
     branches: [ "main" ]
 
 
 jobs:
-  eslint:
-    name: Run eslint scanning
+  build:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
-      - name: Install ESLint
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18.7.0
+      - name: install dependencies
+        working-directory: ./skin-resque-frontend
         run: |
-          npm install eslint@8.10.0
-          npm install @microsoft/eslint-formatter-sarif@2.1.7
-
-      - name: Run ESLint
-        run: npx eslint .
-          --config .eslintrc.js
-          --ext .js,.jsx,.ts,.tsx
-          --format @microsoft/eslint-formatter-sarif 
-          --output-file eslint-results.sarif
-        continue-on-error: true
-
-      - name: Upload analysis results to GitHub
-        uses: github/codeql-action/upload-sarif@v2
+          npm cache clean
+          rm -rf node_modules
+          npm install
+  eslint:
+    name: Run eslint
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
         with:
-          sarif_file: eslint-results.sarif
-          wait-for-processing: true
+          node-version: 18.7.0
+      - name: install dependencies
```

Comment:
Why do we install dependencies again here? Do we need dependencies for static code analysis?